src/python/pants/backend/python/lint/yapf/rules_integration_test.py

```diff
@@ -0,0 +1,261 @@
+        }
+    )
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(
+        rule_runner, [tgt], extra_args=["--yapf-args='--no-local-style'"]
+    )
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original", "f.py"))
+    # by default, PEP8 style is used when no config files are available
+    assert fmt_result.output == get_digest(
+        rule_runner, {"f.py": FIXED_NEEDS_FORMATTING_FILE_INDENT4}
+    )
+    assert fmt_result.did_change is True
+
+
+def test_passthrough_args(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"f.py": NEEDS_FORMATTING_FILE, "BUILD": "python_library(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(
+        rule_runner, [tgt], extra_args=["--yapf-args=--style='{indent_width: 4}'"]
+    )
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original", "f.py"))
+    assert fmt_result.output == get_digest(
+        rule_runner, {"f.py": FIXED_NEEDS_FORMATTING_FILE_INDENT4}
+    )
+    assert fmt_result.did_change is True
+
+
+def test_skip(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"f.py": BAD_FILE, "BUILD": "python_library(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(rule_runner, [tgt], extra_args=["--yapf-skip"])
+    assert not lint_results
+    assert fmt_result.skipped is True
+    assert fmt_result.did_change is False
+
+
+@pytest.mark.parametrize("path,contents,extra_args", ((".yapfignore", "d*py", []),))
+def test_ignore_files(
+    rule_runner: RuleRunner, path: str, contents: str, extra_args: list[str]
+) -> None:
+    """yapf should ignore files specified in the .yapfignore file."""
+    rule_runner.write_files(
+        {
+            "f.py": NEEDS_FORMATTING_FILE,
+            "d.py": NEEDS_FORMATTING_FILE,
+            "BUILD": "python_library(name='t', interpreter_constraints=['==3.9.*'])",
+            path: contents,
+        }
+    )
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(rule_runner, [tgt], extra_args=extra_args)
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original", "f.py"))
+    assert "d.py" not in lint_results[0].stdout
+    assert fmt_result.output == get_digest(
+        rule_runner, {"f.py": FIXED_NEEDS_FORMATTING_FILE_INDENT4}
+    )
+    assert fmt_result.did_change is True
+
+
+def test_ignore_files_empty_yapfignore(rule_runner: RuleRunner) -> None:
```

Comment:
Ditto that this feels more like a test for Yapf itself than for Pants's functionality. Wdyt?

Reply:
Right, it does take a few seconds to run the test and indeed it's more yapf specific, fully agree. I've removed this one.